bindgen/clang.rs

```diff
@@ -496,6 +498,44 @@ impl Cursor {
         }
     }
 
+    /// Traverse this curser's children, sorted by where they appear in source code.
+    ///
+    /// Call the given function on each AST node traversed.
+    pub(crate) fn visit_sorted<Visitor>(
+        &self,
+        ctx: &mut BindgenContext,
+        mut visitor: Visitor,
+    ) where
+        Visitor: FnMut(&mut BindgenContext, Cursor) -> CXChildVisitResult,
+    {
+        let mut children = self.collect_children();
+
+        for child in &children {
+            if child.kind() == CXCursor_InclusionDirective {
+                if let Some(included_file) = child.get_included_file_name() {
+                    let location = child.location();
+
+                    let (source_file, _, _, offset) = location.location();
+
+                    if let Some(source_file) = source_file.name() {
+                        ctx.add_include(source_file, offset, included_file);
+                    }
+                }
+            }
+        }
+
+        children.sort_by(|child1, child2| {
```

Comment:
This might not be super important but what happens if `child1` is included in `child2` via another file?

Reply:
Not quite sure I understand your question. If `child1` is included in `child2` via `child3`, it will be sorted first relative to `child3`, and then relative to `child2`.


Example:

`header1.h`

```c
#define VAR1 1

#include "header2.h"

#define VAR2 2
```

`header2.h`

```c
#define VAR3 3

#include "header3.h"

#define VAR4 4

```

`header3.h`

```c
#define VAR5 5
```

will generate 

```rust
pub const VAR1: u32 = 1;
pub const VAR3: u32 = 3;
pub const VAR5: u32 = 5;
pub const VAR4: u32 = 4;
pub const VAR2: u32 = 2;
```

Reply:
Yeah I think I got entangled by how the ordering works but I get it now, thanks!